cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringAgentService.java

```diff
@@ -61,26 +71,28 @@
 import java.net.URISyntaxException;
 import java.net.URLEncoder;
 import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 import java.util.zip.GZIPInputStream;
-import javax.inject.Inject;
 import javax.inject.Named;
 
 /**
  * The main class to run the remote agent service.
  */
-public class TetheringAgentService extends AbstractRetryableScheduledService {
+public class TetheringAgentService extends AbstractNotificationSubscriberService {
   private static final Logger LOG = LoggerFactory.getLogger(TetheringAgentService.class);
   private static final Gson GSON = ApplicationSpecificationAdapter.addTypeAdapters(new GsonBuilder())
     .registerTypeAdapter(Arguments.class, new ArgumentsCodec())
     .registerTypeAdapter(ProgramOptions.class, new ProgramOptionsCodec())
     .create();
   private static final String CONNECT_CONTROL_CHANNEL = "/v3/tethering/controlchannels/";
   private static final String SUBSCRIBER = "tether.agent";
+  private static final String RUNTIME_SUBSCRIBER = "runtime";
```

Comment:
should we use this constant in RuntimeProgramStatusSubscriberService? The subscriber name is currently hardcoded there

Reply:
added the constant + explanation for use case